WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderThumbnailStrip.java

```diff
@@ -58,79 +57,64 @@ public ReaderThumbnailStrip(Context context, AttributeSet attrs, int defStyleAtt
     }
 
     private void initView(Context context) {
-        mView = inflate(context, R.layout.reader_thumbnail_strip, this);
-        mContainer = (LinearLayout) mView.findViewById(R.id.thumbnail_strip_container);
-        mThumbnailSize = context.getResources().getDimensionPixelSize(R.dimen.reader_thumbnail_strip_image_size);
-        mCountStr = context.getResources().getString(R.string.reader_label_image_count_multi);
+        mView = (ViewGroup) inflate(context, R.layout.reader_thumbnail_strip, this);
+        mThumbnailHeight = context.getResources().getDimensionPixelSize(R.dimen.reader_thumbnail_strip_image_height);
 
-        // base max image count on display width
         int displayWidth = DisplayUtils.getDisplayPixelWidth(context);
-        if (displayWidth <= 800) {
-            mMaxImageCount = 2;
-        } else if (displayWidth <= 1024) {
-            mMaxImageCount = 3;
-        } else if (displayWidth <= 1440) {
-            mMaxImageCount = 4;
-        } else {
-            mMaxImageCount = 5;
-        }
+        int margins = context.getResources().getDimensionPixelSize(R.dimen.reader_card_content_padding) * 2;
+        mThumbnailWidth = (displayWidth - margins) / IMAGE_COUNT;
     }
 
-    public void loadThumbnails(long blogId, long postId, final boolean isPrivate) {
+    public void loadThumbnails(long blogId, long postId, boolean isPrivate) {
         // get rid of any views already added
-        mContainer.removeAllViews();
+        mView.removeAllViews();
 
         // get this post's content and scan it for images suitable in a gallery
         final String content = ReaderPostTable.getPostText(blogId, postId);
         final ReaderImageList imageList =
-                new ReaderImageScanner(content, isPrivate).getGalleryImageList();
-        if (imageList.size() < MIN_IMAGE_COUNT) {
+                new ReaderImageScanner(content, isPrivate).getImageList(IMAGE_COUNT, ReaderConstants.MIN_GALLERY_IMAGE_WIDTH);
```

Comment:
OK, so I am not  completely sure if I understand the design here. So, let me know if I am misunderstanding. Here goes:

`getImageList` takes a `maxImageCount` parameter, so it returns a list with at most `maxImageCount` elements. But then we check if the list is less than the `maxImageCount` we passed in. So basically we are checking if we get exactly a certain amount of images. So far so good.

If this is all we are doing with that method, which seems to be the case considering only other usage of it just passes `0`, wouldn't it better to take `minImageCount` as parameter instead. This way we could refactor the `hasUsableImageCount` method to just use the `getImageList` and return the size of the list? The performance hit we'll take shouldn't be a big deal and should be worth to avoid doing the carbon copy methods.

What do you think?